### PR TITLE
Issue 18051

### PR DESCRIPTION
**Pull request by erickgonzalez**
Now we ship the `CONTENT_ALLOW_ZERO_LENGTH_FILES = true`.

Tests for both possible conditions (true or false) have been created when generating a bundle with a zero bytes file.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java [hunk 1]
```diff
@@ -597,5 +605,80 @@ public void Test_GenerateBundle_PageWithTemplateDeleted_PageSuccessfullyAddedToB
...
+
+    }
+
+    /**
+     * Given Scenario: If a file asset has zero bytes and CONTENT_ALLOW_ZERO_LENGTH_FILES set to false,
+     *                    when you try to generate a bundle with that file asset it fails.
+     * ExpectedResult: DotBundleException should be thrown when trying to generate the bundle with a file with zero bytes.
+     */
+    @Test (expected = DotBundleException.class)
+    public void Test_GenerateBundle_FileAssetZeroBytes_ContentAllowZeroLengthFilesSetToFalse_DotBundleException() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:654]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java [hunk 2]
```diff
@@ -597,5 +605,80 @@ public void Test_GenerateBundle_PageWithTemplateDeleted_PageSuccessfullyAddedToB
 
     }
 
+    /**
+     * Given Scenario: If a file asset has zero files and CONTENT_ALLOW_ZERO_LENGTH_FILES set to true,
+     *                  we should be able to generate the bundle without issues.
+     * ExpectedResult: The bundle should be generated without issues and the page added to the bundle.
+     */
+    @Test
+    public void Test_GenerateBundle_FileAssetZeroBytes_ContentAllowZeroLengthFilesSetToTrue_Success() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:614]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java [hunk 3]
```diff
@@ -597,5 +605,80 @@ public void Test_GenerateBundle_PageWithTemplateDeleted_PageSuccessfullyAddedToB
...
+        final boolean allowZeroLengthFilesDefault = Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", true);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:615]: Codacy Issue found: [The String literal "CONTENT_ALLOW_ZERO_LENGTH_FILES" appears 6 times in this file; the first occurrence is on line 615](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:654]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:614]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/PushPublishBundleGeneratorTest.java:615]: Codacy Issue found: [The String literal "CONTENT_ALLOW_ZERO_LENGTH_FILES" appears 6 times in this file; the first occurrence is on line 615](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5701517)

**Review by jdotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**